lib/DidJwtStrategy.js

```diff
@@ -0,0 +1,204 @@
+  // this allows custom document loaders to be configured
+  didio.use('jsonld', jsonld);
+});
+
+/**
+ * Creates a new didJwt Strategy for use with passport.
+ */
+class Strategy extends passport.Strategy {
+  constructor(options) {
+    super();
+    this.name = 'didJwt',
+    this.options = options;
+    this.store = null;
+  }
+}
+
+module.exports = Strategy;
+
+/**
+ * Authenticate a request.
+ *
+ * @param req the request to authenticate.
+ */
+Strategy.prototype.authenticate = function(req) {
+  const self = this;
+  if(!(`authorization` in req.headers)) {
+    return self.error(new BedrockError(
+      '`Authorization` header is required.', 'InvalidHeader'));
+  }
+  const authHeader = req.headers.authorization.split(' ');
+  if(authHeader.length !== 2 || authHeader[0].toLowerCase() !== 'bearer') {
+    return self.error(new BedrockError(
+      '`Authorization` header format must be `Bearer <token>`.',
+      'InvalidHeader', {
+        header: req.headers.authorization
+      }));
+  }
+  const token = authHeader[1];
+  let decodedToken;
+  let payload;
+  let header;
+  try {
+    decodedToken = jwt.decode(token, {complete: true});
+    payload = decodedToken.payload;
+    header = decodedToken.header;
+  } catch(err) {
+    return self.error(err);
+  }
+
+  if(!('sub' in payload)) {
+    return self.error(new BedrockError(
+      'Token `sub` is required.', 'InvalidToken', {
+        token: decodedToken
+      }));
+  }
+  const did = payload.sub;
+  if(did.indexOf('did:') !== 0) {
+    return self.error(new BedrockError(
+      'Token `sub` must be a valid decentralized identifier.', 'InvalidToken', {
+        token: decodedToken
+      }));
+  }
+
+  let verify;
+  if(header.alg.substring(0, 2) === 'RS') {
+    verify = self.verifyRsa;
+  } else if(header.alg.substring(0, 2) === 'HS' && self.store) {
+    verify = self.store.verify.bind(self.store);
+  } else {
+    return self.error(new BedrockError(
+      'Unsupported signing algorithm.', 'InvalidToken', {
+        token: decodedToken
+      }));
+  }
+
+  verify(token, err => {
+    if(err) {
+      return self.error(err);
+    }
+    req.user = {
+      identity: {id: did}
+    };
+    self.success(req.user);
+  });
+};
+
+Strategy.prototype.setStore = function(store, callback) {
+  this.store = store;
+  async.parallel([
+    callback => {
+      // the store does not handle verification of RSA signed tokens
+      const opts = bedrock.util.clone(config['authn-did-jwt'].namespaces.rsa);
+      delete opts.verify;
+      store.provision('RSA', opts, callback);
+    },
+    callback => store.provision(
+      'HMAC', config['authn-did-jwt'].namespaces.hmac, callback)
+  ], callback);
+};
+
+/**
+ * Verify a JWT signed with a RSA algorithm.
+ *
+ * @param token the token to be verified.
+ *
+ * @param callback(err, tokenPayload) called once the operation completes.
+ */
+Strategy.prototype.verifyRsa = function(token, callback) {
```

Comment:
This should just be `verify`. We can check the token algorithm inside verify to determine if we should use the store or some other method. We shouldn't make the public API so specific like `verifyRsa` for a variety of reasons, but one that springs to mind immediately is that we can use ECDSA w/the same verification technique as RSA once we fully support ECC-based keys.